File: LeetCode/Medium/0647-palindromic-substrings/0647-palindromic-substrings.cpp
class Solution {
public:
    int expand(string &s,int l ,int r){
        int ans=0;
        while(l>=0 && r <s.size() && s[l] == s[r]){
            ans++;
            l--;
            r++;
        }
        return ans;
    }
    int countSubstrings(string s) {
        int ans=0;
        for(int i=0;i<s.size();i++){
            ans += expand(s,i,i);
            ans += expand(s,i,i+1);
        }
        return ans;
    }
};

File: LeetCode/Medium/0647-palindromic-substrings/0647-palindromic-substrings.py
from functools import lru_cache
class Solution:
    def countSubstrings(self, s: str) -> int:
        @lru_cache(maxsize = None)
        def isp(i: int,j: int) -> bool:
            if i>=j:
                return True
            if s[i] != s[j]:
                return False
            return isp(i+1,j-1)
        n = len(s)
        count = 0
        for i in range(n):
            for j in range(i,n):
                if isp(i,j):
                    count += 1
        return count
